File: app/src/main/java/com/lucas/omnia/adapters/EventAdapter.java
package com.lucas.omnia.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.lucas.omnia.R;
import com.lucas.omnia.models.Event;

import java.util.List;

public class EventAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private final List<Event> eventList;

    public static class EventViewHolder extends RecyclerView.ViewHolder {
        TextView dateTextView;
        TextView statusLabelTextView;
        TextView typeTextView;
        TextView descriptionTextView;
        TextView nameTextView;
        TextView buildingTextView;
        TextView roomTextView;
        TextView floorTextView;

        EventViewHolder(View itemView) {
            super(itemView);

            dateTextView = itemView.findViewById(R.id.event_tv_date_body);
            statusLabelTextView = itemView.findViewById(R.id.event_tv_status_body);
            typeTextView = itemView.findViewById(R.id.event_tv_type_body);
            descriptionTextView = itemView.findViewById(R.id.event_tv_description_body);
            nameTextView = itemView.findViewById(R.id.event_tv_name_body);
            buildingTextView = itemView.findViewById(R.id.event_tv_building_body);
            roomTextView = itemView.findViewById(R.id.event_tv_room_body);
            floorTextView = itemView.findViewById(R.id.event_tv_floor_body);
        }
    }

    public EventAdapter(List<Event> eventList) {
        this.eventList = eventList;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        return new EventViewHolder(inflater.inflate(R.layout.item_event, parent, false));
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
        Event event = eventList.get(position);
        TextView dateTextView = ((EventViewHolder) viewHolder).dateTextView;
        TextView statusLabelTextView = ((EventViewHolder) viewHolder).statusLabelTextView;
        TextView typeTextView = ((EventViewHolder) viewHolder).typeTextView;
        TextView descriptionTextView = ((EventViewHolder) viewHolder).descriptionTextView;
        TextView nameTextView = ((EventViewHolder) viewHolder).nameTextView;
        TextView buildingTextView = ((EventViewHolder) viewHolder).buildingTextView;
        TextView roomTextView = ((EventViewHolder) viewHolder).roomTextView;
        TextView floorTextView = ((EventViewHolder) viewHolder).floorTextView;

        dateTextView.setText(event.getDate());
        statusLabelTextView.setText(event.getStatus());
        typeTextView.setText(event.getType());
        descriptionTextView.setText(event.getDescription());
        nameTextView.setText(event.getName());
        buildingTextView.setText(event.getBuilding());
        roomTextView.setText(event.getRoom());
        floorTextView.setText(event.getFloor());
    }

    @Override
    public int getItemCount() {
        return eventList.size();
    }
}

File: app/src/main/java/com/lucas/omnia/adapters/EventAdapter.kt
package com.lucas.omnia.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.lucas.omnia.R
import com.lucas.omnia.models.Event

class EventAdapter(private val eventList: List<Event>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    class EventViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var dateTextView: TextView = itemView.findViewById(R.id.event_tv_date_body)
        var statusLabelTextView: TextView = itemView.findViewById(R.id.event_tv_status_body)
        var typeTextView: TextView = itemView.findViewById(R.id.event_tv_type_body)
        var descriptionTextView: TextView = itemView.findViewById(R.id.event_tv_description_body)
        var nameTextView: TextView = itemView.findViewById(R.id.event_tv_name_body)
        var buildingTextView: TextView = itemView.findViewById(R.id.event_tv_building_body)
        var roomTextView: TextView = itemView.findViewById(R.id.event_tv_room_body)
        var floorTextView: TextView = itemView.findViewById(R.id.event_tv_floor_body)

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return EventViewHolder(inflater.inflate(R.layout.item_event, parent, false))
    }

    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, position: Int) {
        val (date, status, type, description, name, building, room, floor) = eventList[position]
        val dateTextView = (viewHolder as EventViewHolder).dateTextView
        val statusLabelTextView = viewHolder.statusLabelTextView
        val typeTextView = viewHolder.typeTextView
        val descriptionTextView = viewHolder.descriptionTextView
        val nameTextView = viewHolder.nameTextView
        val buildingTextView = viewHolder.buildingTextView
        val roomTextView = viewHolder.roomTextView
        val floorTextView = viewHolder.floorTextView
        dateTextView.text = date
        statusLabelTextView.text = status
        typeTextView.text = type
        descriptionTextView.text = description
        nameTextView.text = name
        buildingTextView.text = building
        roomTextView.text = room
        floorTextView.text = floor
    }

    override fun getItemCount(): Int {
        return eventList.size
    }
}